### PR TITLE
[BD-6] OEP-18 Compliance

### PR DESCRIPTION
**Pull request by ericfab179**
Update setup.py install_requirements definition.
Update travis file using new requirements.
Update tox to use requirement files.
create pip_tools, base, quality, test, tox and travis requirements files.
Add makefile upgrade command.
Include requirements files generated using upgrade command.
Add openedx.yml.
Add MANIFEST.in.
Fix esslint version to avoid problem with version 7.

**Reviewers**
- [ ] @awais786 
- [ ] @morenol

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @ericfab179! I've created [OSPR-4577](https://openedx.atlassian.net/browse/OSPR-4577) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

**Comment by edx-webhook** [comment]
@ericfab179 Even though your pull request wasn’t merged, please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/xblock-free-text-response&entry.867055334=https://github.com/edx/xblock-free-text-response/pull/5&entry.1484655318=https://github.com/ericfab179&entry.752974735=2020-05-22+00:06&entry.1917517419=2020-05-22+00:26&entry.2133058324=No) so we can improve your experience in the future.

File: tox.ini [hunk 1]
```diff
@@ -35,14 +37,14 @@ skip_install = True
 [testenv:pycodestyle]
```
Comment (mraarif) [inline tox.ini:37]: (a suggestion here) why don't we make this a single Tox environment named `quality` that runs both `pycodestyle` and `pylint` while we're at it

**Comment by edx-webhook** [comment]
@ericfab179 Even though your pull request wasn’t merged, please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/xblock-free-text-response&entry.867055334=https://github.com/edx/xblock-free-text-response/pull/5&entry.1484655318=https://github.com/ericfab179&entry.752974735=2020-05-22+00:06&entry.1917517419=2020-05-22+22:58&entry.2133058324=No) so we can improve your experience in the future.

**Comment by awais786** [comment]
@ericfab179 Please squash and bump the version.

**Comment by edx-webhook** [comment]
@ericfab179 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/xblock-free-text-response&entry.867055334=https://github.com/edx/xblock-free-text-response/pull/5&entry.1484655318=https://github.com/ericfab179&entry.752974735=2020-05-22+00:06&entry.1917517419=2020-05-27+10:16&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**